datumaro/components/visualizer.py

```diff
@@ -232,7 +232,18 @@ def vis_one_sample(
         img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
         ax.imshow(img)
 
-        ax.set_title(f"ID: {id}, Subset={subset}")
+        width = ax.transAxes.transform_point((1, 0))[0] - ax.transAxes.transform_point((0, 0))[0]
+        text = ax.set_title(f"ID: {id}, Subset: {subset}", loc="center", wrap=True)
+        text.__get_wrapped_text = text._get_wrapped_text
+
+        def _get_wrapped_text():
+            wrapped_text = text.__get_wrapped_text()
+            text._text = wrapped_text
+            return wrapped_text
+
+        text._get_wrapped_text = _get_wrapped_text
```

Comment:
why this is required? what is the use for `text._text`, `text._get_wrapped_text`, and `text.__get_wrapped_text`?

Reply:
This is fundamentally because `matplotlib` doesn't support this functionality we need.

`matplotlib.pyplot.Text` draw its text with the wrapped text. But it doesn't update its layout with the wrapped text. The layout can only be updated by `Text._text`.

https://github.com/matplotlib/matplotlib/blob/fbfa28d522801dca0aa4e8d688085e4e900cc136/lib/matplotlib/text.py#L751-L753

This un-updated layout makes the text box cropped if it is wrapped so that has multi-lines. This is because its `._text` is a single line text, so that its layout will only cover an area required for the single line. Please see the following table.

| What we want | Un-updated layout | 
|-----------------|-----------------------|
| ![image](https://user-images.githubusercontent.com/26541465/204222709-182d2451-eae6-4c85-9d1b-f8b1fceefa33.png) | ![image](https://user-images.githubusercontent.com/26541465/204222896-ff969dec-f30b-419f-96f9-a9a8be8b82bc.png) |

To prevent this `matplotlib.pyplot.Text` behavior, we can only solve this by a dirty way, letting `_get_wrapped_text()` update `._text` also.

Reply:
Thank you for clear explanation!